### PR TITLE
Recovery email

### PR DESCRIPTION
**Pull request by BigBy996**
image
image

### CODE REVIEW

File: features/page/loginPage.js [hunk 1]
```diff
@@ -9,24 +10,32 @@ const expandAccountOnEnterPasswordSelector = 'div[jscontroller="VXdfxd"]';
...
+const verificationCodeButtonSelector = '.RveJvd.snByac';
+const inputVerificationCodeSelector = 'input[type="tel"]';
+const accountsSelector = '.TnvOCe.k6Zj8d.XraQ3b';
+// Selectors yandex
+const yLoginPageButtonSelector = '.new-hr-auth-Form_Button-enter';
+const yInputEmailLoginPageSelector = 'input[name="login"]';
+const yInputPasswordLoginPageSelector = 'input[name="passwd"]';
+const yEnterLoginPasswordPageButtonSelector = '.passport-Button-Text';
+const yMessagesSelector = '.mail-MessageSnippet-Item_body';
+const yVerificationCodeSelector = 'strong[style="text-align:center;font-size:24px;font-weight:bold;"]';
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: yo...very hard selector. can we change it?

File: features/page/loginPage.js [hunk 2]
```diff
@@ -47,37 +56,90 @@ const login = Object.create({
...
+    login.waitForPageChanged();
+    browser.addValue(firstNameSelector, user.firstName);
+    browser.addValue(lastNameSelector, user.lastName);
+    login.clickNextButton();
+    login.waitForPageChanged();
+    uiUtils.clickWhenVisibleSelector(verificationCodeButtonSelector);
+  },
+
+  getVerificaionCodeFromRecoveryUser: function getVerificaionCodeFromRecoveryUser(user) {
+    browser.newWindow(user.urlAddress);
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: user and url adress - it;s different entity. please create separate object for it

File: features/page/loginPage.js [hunk 3]
```diff
@@ -47,37 +56,90 @@ const login = Object.create({
...
+    uiUtils.addValueWhenVisibleSelector(yInputEmailLoginPageSelector, user.name);
+    uiUtils.addValueWhenVisibleSelector(yInputPasswordLoginPageSelector, user.password);
+    uiUtils.clickWhenVisibleSelector(yEnterLoginPasswordPageButtonSelector);
+  },
+
+  getVerificationCodeYandex: function getVerificationCodeYandex() {
+    browser.waitForVisible(yMessagesSelector);
+    const messages = browser.elements(yMessagesSelector).value;
+    const googleVerificationCodeText = 'Google Verification Code';
+    let i = 0;
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: dont' use counter outside cycli. it's dirty hack

File: features/page/loginPage.js [hunk 4]
```diff
@@ -9,24 +10,32 @@ const expandAccountOnEnterPasswordSelector = 'div[jscontroller="VXdfxd"]';
 const homePageSelector = '.nH.bkL';
 const useAnotherAccountButtonSelector = 'div[id="identifierLink"]';
 const recoveryEmailPhoneNumberTextFieldSelector = 'input[id="recoveryIdentifierId"]';
+const firstNameSelector = 'input[id="firstName"]';
+const lastNameSelector = 'input[id="lastName"]';
+const verificationCodeButtonSelector = '.RveJvd.snByac';
+const inputVerificationCodeSelector = 'input[type="tel"]';
+const accountsSelector = '.TnvOCe.k6Zj8d.XraQ3b';
+// Selectors yandex
+const yLoginPageButtonSelector = '.new-hr-auth-Form_Button-enter';
```
Comment (AleksandrHorev) [inline features/page/loginPage.js]: please create new POM for yandex. we use POM conceptions

**Comment by BigBy996** [comment]
image

File: features/page/homePage.js [hunk 5]
```diff
@@ -13,10 +13,10 @@ const sendMessageButtonSelector = 'div[role="button"]';
...
 const tableRowSelector = 'tr';
 const nextPageButtonsSelector = '.T-I.J-J5-Ji.amD.T-I-awG.T-I-ax7.T-I-Js-Gs.L3';
-const inboxLabelSelector = '.TO.nZ.aiq';
-const outboxLabelSelector = '.TN.GLujEb.aHS-bnu';
-const markedLabelSelector = '.TN.GLujEb.aHS-bnw';
-const draftLabelSelector = '.TN.GLujEb.aHS-bnq';
+const inboxLabelSelector = 'a[href="https://mail.google.com/mail/u/0/#inbox"]';
+const markedLabelSelector = 'a[href="https://mail.google.com/mail/u/0/#starred"]';
+const outboxLabelSelector = 'a[href="https://mail.google.com/mail/u/0/#sent"]';
+const draftLabelSelector = 'a[href="https://mail.google.com/mail/u/0/#drafts"]';
```
Comment (AlexZavkn) [inline features/page/homePage.js]: Maybe we can use a template here?

Comment (AleksandrHorev) [reply features/page/homePage.js]: ye, good point

File: features/step_definitions/login.js [hunk 6]
```diff
@@ -3,23 +3,23 @@ const settings = require('./../../settings.js');
...
   });
 
   this.When(/^I enter email box password to password input on login page$/, function enterPassword() {
-    login.enterPassword(settings.mailUser.password);
+    login.enterPassword(settings.google.user.password);
   });
 
   this.Given(/^I logged to gmail$/, function loginAsUser() {
-    login.loginAs(settings.mailUser, settings.gmailAddress);
+    login.loginAs(settings.google, settings.yandex);
```
Comment (AleksandrHorev) [inline features/step_definitions/login.js]: m...why settings.yandex?

File: settings.js [hunk 7]
```diff
@@ -1,8 +1,23 @@
...
+    },
+  },
+  yandex: {
+    urlAddress: 'https://mail.yandex.ru',
+    user: {
+      name: 'akvmailer@ya.ru',
+      password: '}t{MkcR2xMx|ntT',
+      firstName: 'Akvelon',
+      lastName: 'RecoveryMail',
+      secretAnswer: 'Khorev',
```
Comment (AleksandrHorev) [inline settings.js]: need change. bad practice use real name

File: features/step_definitions/login.js (hunk 6 above)
Comment (BigBy996) [reply features/step_definitions/login.js]: Because we need user and url. They're part of yandex object

**Pull request merged**